src/server.ts

```diff
@@ -95,15 +157,15 @@ class Server extends EventEmitter {
         this.connections = new Map();
     }
 
-    log(connectionId, str) {
+    log(connectionId: unknown, str: string): void {
         if (this.verbose) {
             const logPrefix = connectionId ? `${connectionId} | ` : '';
             console.log(`ProxyServer[${this.port}]: ${logPrefix}${str}`);
         }
     }
 
-    onClientError(err, socket) {
-        this.log(socket.proxyChainId, `onClientError: ${err}`);
+    onClientError(err: NodeJS.ErrnoException, socket: net.Socket): void {
+        this.log((socket as any).proxyChainId, `onClientError: ${err}`);
```

Comment:
maybe we should use own extension of `net.Socket` (interface), this type casting because of `proxyChainId` is quite common